subsys/net/lib/prometheus/counter.c

```diff
@@ -14,15 +14,37 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(pm_counter, CONFIG_PROMETHEUS_LOG_LEVEL);
 
-int prometheus_counter_inc(struct prometheus_counter *counter)
+int prometheus_counter_add(struct prometheus_counter *counter, uint64_t value)
 {
-	if (!counter) {
+	if (counter == NULL) {
 		return -EINVAL;
 	}
 
-	if (counter) {
-		counter->value++;
+	counter->value += value;
```

Comment:
I guess we may check the `value`  if under ZERO so Prometheus document refer following:

```
[Counter](https://prometheus.io/docs/concepts/metric_types/#counter) is a monotonically increasing counter. It MUST NOT allow the value to decrease, however it MAY be reset to 0 (such as by server restart).

A counter MUST have the following methods:

    inc(): Increment the counter by 1
    inc(double v): Increment the counter by the given amount. MUST check that v >= 0.

A counter is ENCOURAGED to have:

A way to count exceptions throw/raised in a given piece of code, and optionally only certain types of exceptions. This is count_exceptions in Python.

Counters MUST start at 0.
```

[Writing client libraries](https://prometheus.io/docs/instrumenting/writing_clientlibs/#counter)

Reply:
Indeed we should, will fix it.

Reply:
Actually, the counter value to add is unsigned so only >= 0 values are possible.